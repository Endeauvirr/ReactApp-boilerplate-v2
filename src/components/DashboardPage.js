import React from 'react';

const DashboardPage = (props) => {
  return (
    <div className="dashboard__wrapper">
      Dashboard page content
    </div>
  );
};

export default DashboardPage;
