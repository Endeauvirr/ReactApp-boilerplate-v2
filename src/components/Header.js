import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ photoURL, displayName, startLogout }) => (
  <header className="loggedUser__header--wrapper">
    <div className="header--nav-section">
      <h1 className="header--logo">
        <NavLink to="/dashboard">
          Boilerplate App
        </NavLink>
      </h1>
      <ul className="header--navigation">
        <li>
          <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        </li>
      </ul>
    </div>
    <div className="header--user-section">
      <div className="user-section__photo">
        <img src={photoURL} alt={displayName} />
      </div>
      <div className="user-section__name--wrapper">
        <p className="user-section__name--label">Logged as:</p>
        <p className="user-section__name">{displayName}</p>
      </div>
      <div className="user-section__menu">
        <button onClick={startLogout}>Log out</button>
      </div>
    </div>
  </header>
);

const mapStateToProps = (state) => ({
  photoURL: state.auth.photoURL,
  displayName: state.auth.displayName
});

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
