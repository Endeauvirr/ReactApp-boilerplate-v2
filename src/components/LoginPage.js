import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
  return (
    <section className="login-page">
      <header className="login-page__header--wrapper">
        <h1>Boilerplate React App</h1>
      </header>
      <div className="login-page__login-panel">
        <div className="login-panel__inner-wrapper">
          <h2>Login</h2>
          <p>Tagline for login</p>
          <button
            className="button-primary"
            onClick={startLogin}
          >
            Authenticate me via Google
          </button>
        </div>
      </div>
    </section>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
