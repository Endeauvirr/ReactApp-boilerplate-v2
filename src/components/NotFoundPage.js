import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    <p>This is not the page you're looking for <span role="img" aria-label="oh noes 404">ðŸ˜±</span></p>
    <p>Go to <Link to="/">Main Page</Link></p>
  </div>
);

export default NotFoundPage;
