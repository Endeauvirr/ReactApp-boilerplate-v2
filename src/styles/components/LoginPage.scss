.login-page {
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 80px calc(100vh - 80px);

  &__header--wrapper {
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background: #fff;
    border-bottom: 1px solid $color-gray;

    h1 {
      margin: 0;
    }
  }

  &__login-panel {
    background: $color-light-gray;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
  }

  .login-panel {
    &__inner-wrapper {
      background: #fff;
      padding: 30px;
      text-align: center;
      box-shadow: 0 10px 20px 0 rgba(0,0,0,0.25);

      h2 {
        margin: 0 0 20px 0;
      }

      p {
        margin: 15px 0;
      }

      button {
        margin: 30px auto 0 auto;
      }
    }
  }
}