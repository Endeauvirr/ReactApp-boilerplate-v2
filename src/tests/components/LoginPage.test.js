import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<LoginPage />);
});

test('Should LoginPage render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => { // Used with spies
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);

  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});
