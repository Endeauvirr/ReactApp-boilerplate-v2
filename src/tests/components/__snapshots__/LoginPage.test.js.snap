// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LoginPage render correctly 1`] = `
<section
  className="login-page"
>
  <header
    className="login-page__header--wrapper"
  >
    <h1>
      Boilerplate React App
    </h1>
  </header>
  <div
    className="login-page__login-panel"
  >
    <div
      className="login-panel__inner-wrapper"
    >
      <h2>
        Login
      </h2>
      <p>
        Tagline for login
      </p>
      <button
        className="button-primary"
      >
        Authenticate me via Google
      </button>
    </div>
  </div>
</section>
`;
